### ISSUE TITLE
Consider exposing Lens.Micro.Mtl.Internal.Focusing

### ISSUE THREAD

**Issue by fmthoma** [aelve/microlens#89]
Currently, I see myself unable to write `Zoom` instances for newtype wrappers around a `StateT`. In `lens`, this used to be possible:

```haskell
newtype MyStateT s m = MyStateT (StateT s m)

type instance Zoomed (MyStateT s m) = Focusing m

instance Monad m => Zoom (MyStateT s m) (MyStateT t m) s t where
    zoom l (MyStateT m) = MyStateT (zoom l m)
```

Without `Focusing` being exposed, this is not possible.

(Or is there another way of defining `zoom` instances for newtypes around `StateT` in `microlens`? This is currently blocking me from migrating from `lens` to `microlens`.)

**Comment by neongreen** [comment]
You can write `type instance Zoomed (MyStateT s m) = Zoomed (StateT s m)`. I still should export `Focusing`, I think, but `type instance Zoomed (MyStateT s m) = Zoomed (StateT s m)` is the preferred way.

**Comment by neongreen** [comment]
(Or rather, it seems to be a better way because then you don't depend on a particular detail of how `Zoomed` is implemented for `StateT`.)

**Comment by fmthoma** [comment]
@neongreen Thanks for the hint. I've tried this already, however it requires `-XUndecidableInstances`, which I am quite hesitant to enable. In the original `lens` version of my code, that was the reason for using `Focusing` in the first place. Is the use of `-XUndecidableInstances` safe in this case?

**Comment by neongreen** [comment]
> Is the use of -XUndecidableInstances safe in this case?

I believe it is. `UndecidableInstances` can lead to loops when resolving instances, but in this case `Zoomed (StateT s m)` can't possibly refer to `MyStateT s m`, so it should be safe.

**Comment by neongreen** [comment]
I'm going to close this issue now; I've pushed a commit that exports `Focusing`, but I'd rather wait until more changes accumulate before releasing the next version :) Ping me if you really-really need it.

**Comment by fmthoma** [comment]
> but I'd rather wait until more changes accumulate before releasing the next version :)

Of course, that is reasonable. I'll stick to `UndecidableInstances` for now. Thank you!

**Comment by neongreen** [comment]
Released in microlens-mtl-0.1.11.
